pkgs/applications/version-management/stgit/default.nix

```diff
@@ -40,24 +45,20 @@ python3Packages.buildPythonApplication rec {
     substituteInPlace Documentation/texi.xsl \
       --replace http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd \
                 ${docbook_xml_dtd_45}/xml/dtd/docbook/docbookx.dtd
-
-    cat > stgit/_version.py <<EOF
-    __version__ = "${version}"
-    EOF
   '';
 
   makeFlags = [
     "prefix=${placeholder "out"}"
     "MAN_BASE_URL=${placeholder "out"}/share/man"
     "XMLTO_EXTRA=--skip-validation"
+    "PERL_PATH=${perl}/bin/perl"
   ];
 
-  buildFlags = [ "all" "doc" ];
+  buildPhase = [ "make" "all" ] ++ makeFlags;
 
-  checkTarget = "test";
-  checkFlags = [ "PERL_PATH=${perl}/bin/perl" ];
+  checkPhase = [ "make" "test" ] ++ makeFlags;
 
-  installTargets = [ "install" "install-doc" "install-html" ];
+  installPhase = [ "make" "install" "install-man" "install-html" ] ++ makeFlags;
```

Comment:
This should be kept installTargets. installPhase and checkPhase are multiline strings, not lists.

Reply:
See my comment for checkPhase/checkTargets

Reply:
As this is the recommended way to package Rust programs with custom build processes, I'll stick with installPhase. But I updated it to be a multiline string as you noted.